client/src/modules/dashboard/Venues/components/VenueForm.tsx

```diff
@@ -112,24 +123,65 @@ const VenueForm: React.FC<VenueFormProps> = (props) => {
     country: venue?.country ?? '',
     latitude: venue?.latitude ?? undefined,
     longitude: venue?.longitude ?? undefined,
+    chapter_id: chapterId,
   };
   const {
+    formState: { isDirty },
     handleSubmit,
     register,
-    formState: { isDirty },
   } = useForm<VenueFormData>({
     defaultValues,
   });
 
+  interface Chapter {
+    id: number;
+    name: string;
+  }
+  let adminedChapters: Chapter[] = [];
+  if (isChaptersDropdownNeeded) {
+    const { user } = useAuth();
+    adminedChapters = user?.admined_chapters ?? [];
+  }
+
   return (
     <>
-      {chapter && <Heading>{chapter.name}</Heading>}
       <form
         aria-label={submitText}
         onSubmit={handleSubmit(onSubmit)}
         className={styles.form}
       >
         <VStack>
+          {!isChaptersDropdownNeeded ? (
+            loadingChapter ? (
+              <Text>Loading Chapter</Text>
+            ) : errorChapter || !dataChapter?.chapter ? (
+              <Text>Error loading chapter</Text>
+            ) : (
+              <Heading>{dataChapter.chapter.name}</Heading>
+            )
+          ) : (
+            <FormControl isRequired>
+              <FormLabel>Chapter</FormLabel>
+              <Select
+                {...register('chapter_id' as const, {
+                  required: true,
+                  valueAsNumber: true,
+                })}
+                defaultValue={
+                  chapterId ??
+                  (adminedChapters?.length ? adminedChapters[0].id : undefined)
+                }
```

Comment:
There are a few bits of JS boolean fun here, so I'd make the default value this:

```
defaultValue =chapterId !== -1
      ? chapterId
      : adminedChapters.length
      ? adminedChapters[0].id
      : -1;
```
Otherwise you can end up with the length being the default value or -1 (because both are truthy).

I also ran into the issue with reloads, too.  The problem seems to be that the default value is getting 'stuck' at one of the earlier versions.  I've been going through the docs and it seems like the `defaultValues` get fixed on first render, but I can't see an obvious fix, since we're always going to render multiple times

```

  useEffect(() => {
    resetField('chapter_id', { defaultValue });
  }, [adminedChapters]);
```
seems to work, if you provide it the new default value. Not pretty, but also not buggy (I think!)

Reply:
Nice, thanks. `useEffect` approach seems to deal with both of these.